packages/flutter_tools/test/general.shard/android/android_project_migration_test.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/android/migrations/top_level_gradle_build_file_migration.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/project_migrator.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:test/fake.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+
+void main() {
+  group('Android migration', () {
+    group('migrate the Gradle "clean" task to lazy declaration', () {
+      late MemoryFileSystem memoryFileSystem;
+      late BufferLogger bufferLogger;
+      late FakeAndroidProject project;
+      late File topLevelGradleBuildFile;
+
+      setUp(() {
+        memoryFileSystem = MemoryFileSystem.test();
+        bufferLogger = BufferLogger.test();
+        project = FakeAndroidProject(memoryFileSystem.currentDirectory.childDirectory('android')..createSync());
```

Comment:
nit: make this a named parameter — otherwise it's not evident when reading the call site what the meaning of the parameter is supposed to be

Reply:
Right, done.